wazo_auth/plugins/http/saml_config/http.py

```diff
@@ -0,0 +1,130 @@
+        idp_metadata = self._validate_xml(file)
+        try:
+            config = self._saml_config_service.create_or_update(
+                scoping_tenant.uuid, saml_config, idp_metadata
+            )
+            return saml_config_schema.dump(config), 200
+        except exceptions.APIException as e:
+            raise e
+        except Exception as e:
+            logger.exception(
+                msg=(
+                    "An error occurred while creating/updating SAML config "
+                    f"for tenant {scoping_tenant.uuid}."
+                )
+            )
+            raise SAMLConfigException(
+                500, f'Unexpected error while processing configuration ({e})', '500'
+            )
+
+    @required_acl('auth.backends.saml.delete')
+    def delete(self):
+        scoping_tenant: Tenant = Tenant.autodetect()
+        try:
+            self._saml_config_service.delete(scoping_tenant.uuid)
+            return '', 204
+        except exceptions.APIException as e:
+            raise e
+        except Exception as e:
+            logger.exception(
+                f"An error occurred while deleting SAML config for tenant {scoping_tenant.uuid}."
+            )
+            raise SAMLConfigException(500, f'Internal server error({e})', '500')
+
+
+class SAMLMetadata(AuthResource):
+    def __init__(self, saml_config_service) -> None:
+        self._saml_config_service = saml_config_service
+
+    @required_acl('auth.backends.saml.read')
+    def get(self):
+        scoping_tenant: Tenant = Tenant.autodetect()
+        try:
+            if etree_metadata := self._saml_config_service.get_metadata(
+                scoping_tenant.uuid
+            ):
+                response = make_response(ElementTree.tostring(etree_metadata))
+                response.headers['Content-Type'] = 'application/xml'
+                response.headers['Content-Disposition'] = 'attachment'
+                return response
+        except exceptions.APIException as e:
+            raise e
+        except Exception as e:
+            logger.exception(
+                f"An error occurred while getting SAML metadata for tenant {scoping_tenant.uuid}."
+            )
+            raise SAMLConfigException(500, f'Internal server error({e})', '500')
+
+
+class SAMLAcsUrl(AuthResource):
+    def __init__(self, saml_config_service) -> None:
+        self._saml_config_service = saml_config_service
+
+    @required_acl('auth.backends.saml.read')
+    def get(self):
+        scoping_tenant: Tenant = Tenant.autodetect()
+        try:
+            if url := self._saml_config_service.get_acs_url(scoping_tenant.uuid):
+                return saml_acs_url_schema.dump(url), 200
+        except exceptions.APIException as e:
+            raise e
+        except Exception as e:
+            logger.exception(
+                f"An error occurred while getting SAML ACS URL for tenant {scoping_tenant.uuid}."
```

Comment:
```suggestion
                f"An error occurred while getting SAML ACS URL for tenant %s.", scoping_tenant.uuid
```

as a quirk of the logging system in python stdlib, it is preferred to write format strings using `%s`/ `%d` placeholders, and provide the format values as additional positional arguments after the format string. This allows the logging system to process the format string and the arguments, in principle allowing custom logic to make use of those separate pieces.

 I think this reflects existing conventions at wazo.